test/Microsoft.ML.CpuMath.UnitTests.DotNetCoreApp/UnitTests.cs

```diff
@@ -0,0 +1,451 @@
+
+        [DllImport("CpuMathNative", EntryPoint = "SumSqU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern float NativeSumSqU(/*const*/ float* ps, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "AddU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern void NativeAddU(/*_In_ const*/ float* ps, /*_Inout_*/ float* pd, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "AddSU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern void NativeAddSU(/*_In_ const*/ float* ps, /*_In_ const*/ int* pi, /*_Inout_*/ float* pd, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "AddScaleU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern void NativeAddScaleU(float a, /*_In_ const*/ float* ps, /*_Inout_*/ float* pd, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "AddScaleSU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern void NativeAddScaleSU(float a, /*_In_ const*/ float* ps, /*_In_ const*/ int* pi, /*_Inout_*/ float* pd, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "ScaleU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern void NativeScaleU(float a, /*_Inout_*/ float* pd, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "Dist2"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern float NativeDist2(/*const*/ float* px, /*const*/ float* py, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "SumAbsU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern float NativeSumAbsU(/*const*/ float* ps, int c);
+
+        [DllImport("CpuMathNative", EntryPoint = "MulElementWiseU"), SuppressUnmanagedCodeSecurity]
+        internal unsafe static extern void NativeMulElementWiseU(/*_In_ const*/ float* ps1, /*_In_ const*/ float* ps2, /*_Inout_*/ float* pd, int c);
+
+        [Theory]
+        [InlineData(10, 0, EXP_MAX)]
+        [InlineData(20, 0, EXP_MAX)]
+        [InlineData(20, 1, EXP_MAX / 2)]
+        [InlineData(30, 2, EXP_MAX / 2)]
+        public unsafe void DotUTest(int len, int seed, int expRange)
+        {
+            // arrange
+            float[] src = new float[len];
+            float[] dst = new float[len];
+            Random rand = new Random(seed);
+
+            for (int i = 0; i < len; i++)
+            {
+                src[i] = NextFloat(rand, expRange);
+                dst[i] = NextFloat(rand, expRange);
+            }
+
+            fixed (float* psrc = src)
+            fixed (float* pdst = dst)
+            {
+                // act
+                var nativeOutput = NativeDotU(psrc, pdst, len);
+                var managedOutput = CpuMathUtils.DotProductDense(src, dst, len);
+                output.WriteLine($"{nativeOutput} == {managedOutput}");
+
+                // assert
+                Assert.True(nativeOutput == managedOutput);
```

Comment:
In general, don't use `Assert.True(a == b)`.  Instead use `Assert.Equal(a, b)`.  This way when the test fails, it fails and says "Actual: 3. Expected 4" instead of just a general "Failed" message.

Reply:
Agree and will fix that - thanks!